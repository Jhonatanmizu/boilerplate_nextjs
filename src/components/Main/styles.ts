import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #282a36;
  color: #f2f2f2;
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  flex-direction: column;
  gap: 1.6rem;
  padding: 3rem;
`

export const Title = styled.h1`
  font-size: 4rem;
`

export const Description = styled.h2`
  font-size: 2rem;
`
