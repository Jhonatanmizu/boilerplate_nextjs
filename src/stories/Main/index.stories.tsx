import Main from '../../components/Main'

const storie = {
  title: 'Main',
  component: Main
}

export const Basic = () => <Main />

export default storie
