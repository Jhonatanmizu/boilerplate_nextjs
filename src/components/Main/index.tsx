import * as S from './styles'

const Main = () => {
  return (
    <S.Wrapper>
      <S.Title>Nextjs boilerplate - Mizugawa</S.Title>
      <S.Description>Typescript, Nextjs and Styled components</S.Description>
    </S.Wrapper>
  )
}
export default Main
