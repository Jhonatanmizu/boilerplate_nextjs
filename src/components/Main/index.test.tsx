// JEST
import { render, screen } from '@testing-library/react'
// Components
import Main from '.'

describe('<Main/>', () => {
  it('Should render a heading', () => {
    const { container } = render(<Main />)
    expect(
      screen.getByRole('heading', { name: /Nextjs boilerplate - Mizugawa/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render colors correctly', () => {
    const { container } = render(<Main />)
    expect(container.firstChild).toHaveStyle({
      'background-color': '#282a36'
    })
  })
})
