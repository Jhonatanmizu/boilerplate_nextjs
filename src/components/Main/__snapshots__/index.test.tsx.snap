// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> Should render a heading 1`] = `
.c0 {
  background-color: #282a36;
  color: #f2f2f2;
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.6rem;
  padding: 3rem;
}

.c1 {
  font-size: 4rem;
}

.c2 {
  font-size: 2rem;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Nextjs boilerplate - Mizugawa
  </h1>
  <h2
    class="c2"
  >
    Typescript, Nextjs and Styled components
  </h2>
</main>
`;
